File: sumTwoSmallestNumbers.cpp
// file:	sumTwoSmallestNumbers.cpp
// author:	Diogo Marques
// data:	13/04/2025
// description:
//	Sum of Two Smallest Numbers at Codewars.com
#include <vector>

long sumTwoSmallestNumbers(std::vector<int> numbers)
{
   if(numbers.size() < 4){return 0;}
  long a = numbers[0];
  long b = numbers[1];
  for(int i = 0; i < numbers.size(); i++){
    if(numbers[i] < a){
      b = a;
      a = numbers[i];
    }
    if(numbers[i] < b && numbers[i] > a){
      b = numbers[i];
    }
  }
  return a + b;
}

File: sumTwoSmallestNumbers.py
# file:	sumTwoSmallestNumbers.py
# author:	Diogo Marques
# data:	13/04/2025
# description:
#	Sum of Two Smallest Numbers at Codewars.com
def sum_two_smallest_numbers(numbers):
    if(len(numbers) < 4): 
        return 0
    
    a = numbers[0]
    b = numbers[1]
    
    for i in numbers:
        if(i < a):
            b = a
            a = i
        if(i < b and i > a):
            b = i
    return a + b